provider/hana/hana.go

```diff
@@ -387,7 +394,13 @@ func CreateProvider(config dict.Dicter, maps []provider.Map, providerType string
 		return nil, err
 	}
 
+	name, err := config.String(ConfigKeyName, nil)
+	if err != nil {
+		return nil, err
+	}
+
 	p := Provider{
+		name:      name,
 		dbVersion: uint(majorVersion),
 		srid:      uint64(srid),
 		pool:      &connectionPoolCollector{pool: conn},
```

Comment:
Shouldn't you be passing in the ProviderName to the connectionPoolCollector here?

Reply:
good catch, long week 👍